grails-app/controllers/org/pih/warehouse/user/UserController.groovy

```diff
@@ -234,6 +239,16 @@ class UserController {
 
         def userInstance = User.get(params.id)
         if (userInstance) {
+            // user is not allowed to edit another user's account unless the user has at least an admin role
+            if (authService?.currentUser?.id != params.id && !userService?.isUserAdmin(authService?.currentUser)) {
```

Comment:
@jmiranda Initially I suggested this as a further improvement to the issue here, but it still does not resolve the issue 100%, we still might have some potential problems with unwanted changes made by users to their accounts and their authorization parameters (through the postman or something like that).
I think that the easiest would be to introduce a separate user edit page stripped of the authorization tab for editing personal data and a password (or even only a simple "change password" form with dedicated action) that could be accessed by the user himself and an admin+ user and a separate (current version of) user edit page for admin+ users.

Reply:
What do you think? Or we are moving this into 9.1, and skipping this as a quick hotfix for now, since Manon said in a ticket that it is not as pressing as it was while Lesotho was going live?

Reply:
@awalkowiak Yeah, the separate form / action for changing the password is what I was thinking we could do for now. In fact, I think we should do this regardless of how we implement the record-level authorization rules. 

Regarding the release, I would still move this to 0.9.1 because it doesn't seem urgent enough to hotfix. But that's just my vote. I would check again with Kelsey and Manon to see if they still agree that it can wait until March.

Reply:
We also need to add a ticket to think through how we want to handle the record-level authorization. I actually think this might be possible by adding a ROLE_SELF to the minimum required roles or adding a boolean property to the map to allow the user to edit their own profile and/or password.

Note: I imagine we still want to allow superusers (maybe admins) as well as the user themself to edit their password and update profile information. We also probably need to take user active flag into account.

So I guess for now we don't need to do anything fancy in the interceptor, we can just implement logic in the controller action. But consider delegating the auth logic to a service.
```
userService.canUserEditUser()
```
:shrug: